src/compile/utils.ts

```diff
@@ -212,21 +202,70 @@ export async function compileSol(
     fileName: string,
     version: string | CompilerVersionSelectionStrategy,
     remapping: string[],
+    compilationOutput?: CompilationOutput[],
+    compilerSettings?: any,
+    kind?: CompilerKind
+): Promise<CompileResult>;
+export async function compileSol(
+    fileNames: string[],
+    version: string | CompilerVersionSelectionStrategy,
+    remapping: string[],
+    compilationOutput?: CompilationOutput[],
+    compilerSettings?: any,
+    kind?: CompilerKind
+): Promise<CompileResult>;
+export async function compileSol(
+    input: string | string[],
+    version: string | CompilerVersionSelectionStrategy,
+    remapping: string[],
     compilationOutput: CompilationOutput[] = [CompilationOutput.ALL],
     compilerSettings?: any,
     kind?: CompilerKind
 ): Promise<CompileResult> {
-    const sourceCode = fse.readFileSync(fileName, { encoding: "utf-8" });
+    const fileNames = typeof input === "string" ? [input] : input;
 
-    return compileSourceString(
-        fileName,
-        sourceCode,
-        version,
-        remapping,
-        compilationOutput,
-        compilerSettings,
-        kind
-    );
+    assert(fileNames.length > 0, "There must be at least one file to compile");
+
+    const fsResolver = new FileSystemResolver();
+    const npmResolver = new LocalNpmResolver();
+    const resolvers = [fsResolver, npmResolver];
+    const parsedRemapping = parsePathRemapping(remapping);
+    const files = new Map<string, string>();
+    const visited = new Set<string>();
+
+    for (const fileName of fileNames) {
+        const sourceCode = await fse.readFile(fileName, "utf-8");
+
+        npmResolver.baseDir = path.dirname(fileName);
```

Comment:
This is not a blocker, but I am wondering if there is a situation, where different files might have different `node_modules` directories, and if in such a situation there might be breakage? I can't come up with a failure scenario, just something to consider I guess

Reply:
There may be a scenario when some absolute path was imported and it has own `node_modules`. I'm wondering the correct behavior in that case. What we currently do - is allow to resolve all files, relatively to each possible path. Otherwise it would be hard to pick common directory for different relative/absolute/virtual possible paths. I'm not sure about "correct"/reliable logic here.

Reply:
I think the current behavior is as close as possible to "correct" in this case. If we run into such a case, and it causes problems lets revisit. For now I am down to merge